src/writer.js

```diff
@@ -185,6 +179,29 @@ define([
         });
     };
 
+    var createSetValues = function (dataTree, form, xmlWriter) {
+        function writeSetValue(setValue) {
+            xmlWriter.writeStartElement('setvalue');
+            xmlWriter.writeAttributeString('event', setValue.event);
+            xmlWriter.writeAttributeString('ref', setValue.ref);
+            xmlWriter.writeAttributeString('value', setValue.value);
+            xmlWriter.writeEndElement();
+        }
+
+        _.each(form.getSetValues(), function (setValue) {
+            writeSetValue(setValue);
+        });
+
+        dataTree.walk(function (mug, nodeID, processChildren) {
+            if(mug && mug.options.getSetValues) {
+                _.each(mug.options.getSetValues(mug), function(setValue) {
+                    writeSetValue(setValue);
+                });
+            }
+            processChildren();
+        });
```

Comment:
What you're doing with setvalues in this PR is looking really sweet, but there's a problem here. Setvalue nodes are evaluated in document order. The parser is not keeping track of, and the writer is not preserving the order in which setvalue nodes were originally parsed from the form. That means opening a form and saving it could break it because setvalue nodes would be written out in a different order.

I don't see an easy solution to this problem with the approach you've taken here. I wonder what @ctsims thinks about writing setvalues in a different order than they were parsed? Is it a major issue?